Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1482,6 +1485,18 @@ extension BrowserViewController: TabToolbarDelegate {
         menuViewController = mvc
     }
 
+    private func toggleNoImageModeForTabState() {
+        let noImageMode = self.profile.prefs.boolForKey(noImageModeKey) ?? false
+        self.profile.prefs.setBool(!noImageMode, forKey: noImageModeKey)
+        if (homePanelController != nil) {
+            homePanelController?.isNoImageMode = !noImageMode
+        }
+        NSNotificationCenter.defaultCenter().postNotificationName(NoImageModeStatusChangedNotification, object: nil)
+        if noImageMode == true {
+            self.tabManager.selectedTab?.reload()
+        }
```

Comment:
You're posting a notification (which updates all tabs) and reloading the current one. Don't we just need to call `__firefoxNoImageMode__.setNoImageMode` for the one tab?